mkdocs/contrib/search/search_index.py

```diff
@@ -175,10 +170,17 @@ def __init__(self, *args, **kwargs) -> None:
         self.data: list[ContentSection] = []
         self.section: ContentSection | None = None
         self.is_header_tag = False
+        self.is_permalink = False
         self._stripped_html: list[str] = []
 
     def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None:
         """Called at the start of every HTML tag."""
+        atts = dict(attrs)
```

Comment:
Creating a dict on every tag occurrence is likely to become a performance bottleneck. It should be created only if necessary.

Reply:
I agree. However, we need to retrieve one of the attributes in the first if statement. In other words, we need the dict before we know if we need it. I'm open to suggestions here.

Reply:
Okay, I worked this out.